File: age-calculator/python/src/age_calculator/age_calculator.py
from datetime import date


def calculate(birthdate: date, today: date) -> int:
    if birthdate > today:
        raise ValueError("birthdate is after today")

    years = today.year - birthdate.year
    if (today.month, today.day) < (birthdate.month, birthdate.day):
        years -= 1
    return years

File: age-calculator/csharp/src/AgeCalculator/AgeCalculator.cs
namespace AgeCalculator;

public static class AgeCalculator
{
    public static int Calculate(DateOnly birthdate, DateOnly today)
    {
        if (birthdate > today)
        {
            throw new ArgumentException("birthdate is after today");
        }

        var years = today.Year - birthdate.Year;
        var birthdayNotYetReached =
            today.Month < birthdate.Month ||
            (today.Month == birthdate.Month && today.Day < birthdate.Day);
        return birthdayNotYetReached ? years - 1 : years;
    }
}
